### ISSUE TITLE
Can't build  ParagraphVectors using existing WordVectors

### ISSUE THREAD

**Issue by Antossio** [deeplearning4j/deeplearning4j#2970]
Hi,
I faced the problem during building ParagraphVectors using existing WordVectors.
I used code from example:
ParagraphVectors vectors = new ParagraphVectors.Builder()
                                  .useExistingWordVectors(word2Vec)
				  .tokenizerFactory(t)
				  .build();
But after paragraph vector was built, it doesn't contain lookupTable and vocabCache because following rows in build() method are commented:
  //this.lookupTable = this.existingVectors.lookupTable();
                //this.vocabCache = this.existingVectors.vocab();

So, both fields are null, and when trying to get inferVector it's also null.
Why it's commented and how to build ParagraphVectors using existing WordVectors?

Thank you in advance

**Comment by raver119** [comment]
Good question :/

**Comment by raver119** [comment]
That's not a bug, existing model used there is used/merged during fit() call in paravec.

To use it for inference directly (without calling fit()) - get lookup table & vocab from w2v model, and pass it into your model.

Also, make sure you have proper w2v model, with either hs or negative sampling states available.

**Comment by raver119** [comment]
However, this thing should be more straightforward to use...

**Comment by raver119** [comment]
Can you please show NPE you're getting there?

**Comment by raver119** [comment]
Here's test for existing w2v model btw: https://github.com/deeplearning4j/deeplearning4j/blob/be32238cf522238e9c6ecfc7827f0e0694f1e9bb/deeplearning4j-nlp-parent/deeplearning4j-nlp/src/test/java/org/deeplearning4j/models/paragraphvectors/ParagraphVectorsTest.java#L698-L698

**Comment by Antossio** [comment]
It was not NPE but when trying get vector by calling
 INDArray inferredVectorA2 = vectors.inferVector("some sentence1");
 it return null for any sentence, even if words were present in initial w2v model.
 
It was not obvious that we need call fit() even if we build paragraph vector based on existing w2v model.
Bu now it's clear, thank you for detail explanation.

**Comment by raver119** [comment]
nono, you don't need fit() don't get me wrong. 

Test i've linked above is for special cases, when you're building paravec on limited corpus, but want to have rich vector space for future inference use. 

And you're asking about slightly different use case. I'm working on it today, got it working, but not happy with results yet, so not closing this thread yet.

**Comment by Antossio** [comment]
I try to get similarity of sentences vector base on trained w2v model with following code:
Word2Vec word2Vec = WordVectorSerializer.readWord2VecModel("modelFromFile.txt", true);
TokenizerFactory t = new DefaultTokenizerFactory();
			t.setTokenPreProcessor(new CommonPreprocessor());
ParagraphVectors paraVectors = new ParagraphVectors.Builder()
	        		.tokenizerFactory(t)
	                .useExistingWordVectors(word2Vec)
	                .build(); 
paraVectors.setLookupTable(word2Vec.lookupTable());
paraVectors.setVocab(word2Vec.getVocab());

INDArray inferredVectorA = vectors.inferVector("some sentence 1");
INDArray inferredVectorB = vectors.inferVector("completely different 2");
INDArray inferredVectorC = vectors.inferVector("some similar sentence 2");

It seems that there is no problem with vocab and lookuptable.
But now I faced another issue - for any sentence, method inferVector returns array filled with 0.0 and -0.0.
I looked through the code and in method iterateSampl(...)  of SkipGram class, there is a condition: 
if(w1 == null || lastWord == null || lastWord.getIndex() < 0 || w1.getIndex() == lastWord.getIndex() || w1.getLabel().equals("STOP") || lastWord.getLabel().equals("STOP") || w1.getLabel().equals("UNK") || lastWord.getLabel().equals("UNK"))
            return 0.0;

But lastWord.getIndex() is -1 because earlie following label was set:
sequence.setSequenceLabel(new VocabWord(1.0, String.valueOf(new Random().nextInt())));

so it's always return 0.0, and inferenceVector returns without any changes for any sentence.
